### ISSUE TITLE
Portals

### ISSUE THREAD

**Issue by dennemark** [brianzinn/react-babylonjs#164]
Hi,

I was wondering if it is possible to use the reconciler function to create a portal.
Looking at how react-three-fiber is doing it, it seems to be a simple [function](https://github.com/pmndrs/react-three-fiber/blob/13f37ffe12d1350e671f3f078a81f2c6a03cf666/packages/fiber/src/web/index.tsx#L178)
However, currently the reconciler is not exposed. The createReconciler function lives within Scene and it does not return the reconciler itself.
I managed to get access to it, but it does not seem to be that easy :/ Getting error `Unexpected Fiber popped.` and `Expected host context to exist`

Even if I place the createReconciler function outside of Scene, so that it would be called only once, it leads to the same error.

Would it be straightforward and worth to implement?

For completeness, I was trying to achieve something like:
```
import {useRef} from "react"
import {createPortal} from "react-babylonjs"
const Portal = (props) => {
  const ref = useRef(null)
  const createPortal = usePortal();

  return (<>
    {ref && createPortal(<box color={Color3.Yellow()} position={new Vector3(2, 0, 0)} />, ref)}
    <transformNode name="test" ref={ref} position={new Vector3(0,-1,0)} />
  </>);
}
```

(I just realized, I should have posted it in disccusions....)

**Comment by brianzinn** [comment]
it's better here in issues, because i keep losing track of the discussions and i try to keep issue count down! :)

I will definitely add that.  I see what you mean as the `createReconciler` call is buried in Scene, so then it needs to be moved out.  I'll need to look at how react-three-fiber exposes their reconciler and allows it to bind to multiple scenes in that kind of global way.  Another option is to put it in the context and add a hook that can access it.  I wouldn't mind moving away from context since it has other issues at the same time.  I'm ok with any breaking changes in that area as I am ready for a v4 anyway with some new ideas for tree shaking.  Do you have a code proposal - I will look more at the code tomorrow.   I had looked previously into createPortal for DOM for the canvas already as you may have seen in Engine.tsx.

Thanks for bringing this up, that will be a cool addition.   You have lots of good ideas - wish I had more time to devote to them!

**Comment by brianzinn** [comment]
> I was trying to convert the HTML overlay to babylonjs. Still needs some refinement, I hope I can push it soon.
Yess!! That would be really nice to add as well.  If you want to start a new issue/branch - would be happy to look even at a draft version 😄
